tests/test_formdata.py

```diff
@@ -0,0 +1,36 @@
+"""Test module for requests_toolbelt.utils.formdata."""
+try:
+    from urllib.parse import parse_qs
+except ImportError:
+    from urlparse import parse_qs
+
+from requests_toolbelt.utils.formdata import urlencode
+
+query = {
+    'first_nested': {
+        'second_nested': {
+            'third_nested': {
+                'fourth0': 'fourth_value0',
+                'fourth1': 'fourth_value1',
+            },
+            'third0': 'third_value0',
+        },
+        'second0': 'second_value0',
+    },
+    'outter': 'outter_value',
+}
```

Comment:
This test doesn't actually include a list or a tuple, and I'm willing to bet that if/when you add those types you won't get the end result you expect. =)